README.md

```diff
@@ -1,61 +1,81 @@
 
-More information to come about the required development environment and workflow.
+6- The validators will detect the new event and execute the mutating query in the corresponding table.
 
-## Updating smart contract bindings
+7- The user can query the RPC endpoint of the validator to execute read-queries (e.g: `SELECT * FROM ...`), to see the materialized result of its interaction with the SC.
 
-The `go-tableland` interactions with the registry and NFT smart contracts are done using generated Golang bindings in the `pkg/tables/impl/ethereum` and `pkg/<fill me in>` packages, respectively. The bindings are generated by running `make ethereum`, but rely on the presence of some manually imported/maintained files in each package: `abi.json` and `bytecode.bin`. If anything about the smart contract APIs changes, these files need to be updated, and the Go bindings re-generated. The contents of these files are created by compiling the smart contracts in [eth-tableland](https://github.com/textileio/eth-tableland) and copying some of the resulting build output.
+> 💡 The description above is optimized to understand the general mechanics of the validator. Minting tables, and executing mutating statements also imply more work both at the SC and validator levels (e.g: ACL enforcing); we’re skipping them here.
 
-### Building eth-tableland contracts
+The validator detects the smart contract events using an EVM node API (e.g: `geth` node), which can be self-hosted or served by providers (e.g: Alchemy, Infura, etc).
 
-Check out the latest code from [eth-tableland](https://github.com/textileio/eth-tableland) and follow the instructions in the `README.md` to [setup and compile the contracts](https://github.com/textileio/eth-tableland/blob/main/README.md#building-the-client).
+# How can I run a validator?
 
-Then follow the instructions to [extract the contract ABI and bytecode](https://github.com/textileio/eth-tableland/blob/main/README.md#extacting-the-abi-and-bytecode).
+Soon we'll be creating full documentation on how to configure and run a validator, stay tuned.
+If you're interested in running a Tableland validator, please [fill this form](https://hhueol4i6vp.typeform.com/to/gkcyeA22).
 
-Copy the resulting `abi.json` and `bytecode.bin` files to the corresponding Go package in this repo, and then run `make ethereum`. If the contract API changed at all, you'll have to make the approprite updates to any code that consumes the generated bindings.
+# Building from source
 
-# How to publish a new version
+You can build the validator binary from source by running: `make build-api`.
+
+# Run with docker-compose
+
+The repository contains in the `docker` folder a complete docker-compose setup to run a validator.
+
+Soon we'll publish dedicated documentation with instructions on how to run it.
+
+# Tools
+
+The `cmd/toolkit` is a CLI which contain useful commands:
+
+- `completion`: Generate the autocompletion script for the specified shell
```

Comment:
`completion`? Where is this coming from?

Reply:
Oh, it's a subcommand that COBRA automatically adds. You can see it with `toolkit --help`, I'll remove it from here since can confuse people, but it's a real sub-command.